slateratom/lib/avgpot.f90

```diff
@@ -0,0 +1,450 @@
+!> Module to calculate the approximate average potential and effective orbital energies proposed in:
+!! Baer R, Livshits E, Salzner U. Tuned range-separated hybrids in density functional theory.
+!! Annual review of physical chemistry. 2010 May 5;61:85-109.
+module average_potential
+
+  use common_accuracy, only : dp, mc
+  use common_constants, only : pi
+  use common_message, only : error
+
+  use utilities, only : index_heap_sort
+  use density, only : wavefunction, wavefunction_1st, wavefunction_2nd
+
+  implicit none
+  private
+
+  public :: getAveragePotential
+
+
+contains
+
+
+  !> Tries to infer the energy of the highest occupied atomic orbital (HOAO) and the associated
+  !! principal and angular quantum number from the eigenvalues and occupations handed over.
+  !!
+  !! The effort made here would be an argument for moving this functionality to the sktools
```

Comment:
In principle the sktools have all the information needed (some more even), so the functionality could be moved there. The process of e.g. inferring the HOAO would be less cumbersome I suppose, since already implemented.

Reply:
I decided to just remove the comment, it is admittedly more confusing than helpful.